modules/openthread/platform/radio.c

```diff
@@ -361,6 +361,10 @@ void platformRadioInit(void)
 
 	cfg.event_handler = handle_radio_event;
 	radio_api->configure(radio_dev, IEEE802154_CONFIG_EVENT_HANDLER, &cfg);
+
+#if defined(CONFIG_OPENTHREAD_LINK_METRICS_SUBJECT)
+	otLinkMetricsInit(otPlatRadioGetReceiveSensitivity());
+#endif
```

Comment:
Seems like these changes break building of samples. CC @edmont

Reply:
Please cherry-pick fix: https://github.com/zephyrproject-rtos/zephyr/pull/67616

Reply:
fix in https://github.com/zephyrproject-rtos/zephyr/pull/67616

Reply:
cherry-picked